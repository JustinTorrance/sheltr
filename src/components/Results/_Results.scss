.shelter-list {
  background: white;
  opacity: 0.8;
  position: absolute;
  top: 300px;
  margin-top: 30px;
  border-radius: 20px;
  color: black;
  list-style-type: none;


}

.shelter-li {
  text-decoration: none !important;
  width: 450px;
  height: 60px;
  margin: 5px;
  border: solid 1px black;
  border-radius: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 1.4rem;
  color: black;
    &:hover {
      color: white;
      background-color: black;
      cursor: pointer;
      transition: .2s;
    }

}