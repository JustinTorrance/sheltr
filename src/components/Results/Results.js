import React from 'react'
import { connect } from 'react-redux'
import { Link, Switch, Route } from 'react-router-dom'
import { fetchPetsByShelter } from '../../thunks/fetchPetsByShelter'
import './_Results.scss'

const Results = (props) => {

  const handleClick = async (e) => {
    const clickedShelter = props.shelters.find(shelter => {
      const shelterName = e.target.innerText
      return shelter.name.$t === shelterName
    })
    await props.fetchPetsByShelter(clickedShelter.id.$t)
  }

  const shelters = props.shelters.map(shelter => {
    return  <Link to='/shelter'>
              <li className='shelter-li' onClick={handleClick}>{shelter.name.$t}</li>
            </Link>
  })

  return(
    <ul className='shelter-list'>
       { shelters }
    </ul>
  )
}

export const mapStateToProps = (state) => ({
  shelters: state.shelters
})

export const mapDispatchToProps = (dispatch) => ({
  fetchPetsByShelter: (shelterID) => dispatch(fetchPetsByShelter(shelterID)),
})

export default connect(mapStateToProps, mapDispatchToProps)(Results)