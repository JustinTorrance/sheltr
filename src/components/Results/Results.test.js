import Results from './Results'
import { mapDispatchToProps } from './Results'
import React from 'react';

describe('Results', () => {
  it('matches the snapshot', () => {
    const wrapper = <Results />
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a fetchPetsByShelter thunk when handleClick is called', () => {
      const mockDispatch = jest.fn()
      const thunkToDispatch = fetchPetsByShelter(e)
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.toggleFavorite(e)
      expect(mockDispatch).toHaveBeenCalledWith(thunkToDispatch)
    })
  })
})