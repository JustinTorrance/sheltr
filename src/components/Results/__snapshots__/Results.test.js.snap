// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results matches the snapshot 1`] = `<Connect(Results) />`;
